src/main/scala/se/nimsa/sbx/filtering/FilteringDAO.scala

```diff
@@ -0,0 +1,203 @@
+package se.nimsa.sbx.filtering
+
+import se.nimsa.dicom.data.TagPath
+import se.nimsa.dicom.data.TagPath.TagPathTag
+import se.nimsa.sbx.app.GeneralProtocol.{SourceRef, SourceType}
+import se.nimsa.sbx.filtering.FilteringProtocol._
+import se.nimsa.sbx.util.DbUtil.createTables
+import slick.basic.DatabaseConfig
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class FilteringDAO(val dbConf: DatabaseConfig[JdbcProfile])(implicit ec: ExecutionContext) {
+
+  import dbConf.profile.api._
+
+  val db = dbConf.db
+
+  private val toTagFilter = (id:Long, name: String, tagFilterType: String) => TagFilter(id, name, TagFilterType.withName(tagFilterType))
```

Comment:
Not used much in slicebox unfortunately but a nice-to-know. You can create a mapping of a class such as TagFilterType to a String and then use TagFilterType as a column type directly in the table. See http://slick.lightbend.com/doc/3.2.3/userdefined.html#scalar-types